react-app/src/component/Legend/DiscreteLegend.tsx

```diff
@@ -97,6 +102,20 @@ export const DiscreteColorLegend: React.FC<discreteLegendProps> = ({
       let useSelectorLegend = false;
       let itemName: string[] = [];
       let itemColor: ItemColor[] = [];
+
+      const defaultDiscreteData: any = {
```

Comment:
IMHO the default table should be empty. The consumer of this component should be able populate and set a default table depending on the use case.

So basically just expose this table as a prop which is default empty. Would be nice with a storybook example that demonstrates this. We could implement this incrementally if you like.

Reply:
Yes I agree it is up to the consumer of the component to populate the labels 

So the basic editing of labels is what we need now.
After edit the component and view using that table should be updated.

Reply:
Agree. This defaultDiscreteData was actually I had written for testing purpose, which I forgot to remove basically. Will look into this and make necessary changes